trunk/NDHMS/Routing/Reservoirs/Persistence_Level_Pool_Hybrid_Module_README.md

```diff
@@ -0,0 +1,113 @@
+will read a timeslice file and return a corresponding observed discharge. The timeslice files will be read at a particular update
+time. The first hybrid reservoir on each processor to reach an update time will call the function to read the timeslice files, and
+the resulting observations will be used by each reservoir on that processor. The release function also peforms the functionality
+described above including calling level pool release along with weighting and combining that release with the weighted observed
+discharge, and finally calling mass balance checks before returning the release/outflow back to the model.
+
+The module_persistence_levelpool_hybrid_parameters.F module defines and instantiates objects for a hybrid type reservoir's
+parameters/properties. Parameters holds static/unchanging variables that are set when the given reservoir object is
+initialized/instantiated.
+
+The module_persistence_levelpool_hybrid_state.F module defines and instantiates objects for a hybrid type reservoir's state.
+State holds and tracks dynamic/changing variables that are only relevant to the given hybrid reservoir object and not other
+modules or areas of the system.
+
+The module_persistence_levelpool_hybrid_tests.F module holds unit tests that test for all components of a hybrid reservoir
+are properly initialized.
+
+Within the Reservoirs directory, the module_read_timeslice_data.F module reads USGS timeslice files to get gage discharge
+values that will be used by reservoirs. Future development will either expand this module or replicate its functionality to
+another module to read in other forms of observation and forecast data. An observation lookback period is passed in to
+determine how far back in time from the current model time the module will look for timeslice files. The observation
+resolution determines the time increments the module will look back. For instance, a standard lookback period would be
+18 hours with an observation resolution of 15 minutes, where a model current time of 8:00 PM would search for timeslice
+files at every 15 minute increment between 2:00 AM and 8:00 PM that day. The module will first search for the most recent
+timeslice files and grab the discharge for a particular lake/reservoir if the gage quality standard is met at that time.
+If a gage discharge is missing or if the gage quality standard is not met for any particular lake/reservoir in the given
+timeslice file, the module will continue to look back at every observation resolution increment until either all
+lakes/reservoirs have a good quality discharge or the end of the lookback period is reached. The total lookback seconds
+from current model time that the discharge is read will also be returned.
+
+Also within the Reservoirs directory, the module_reservoir_utilities.F module contains multiple functions used by the Hybrid
+module. The modify_for_projected_storage function is called from the hybrid release function and modifies the outflow if a
```

Comment:
It almost seems like modify_for_projected_storage is unique to the hybrid. 

Generally, this file is giving me a smell. But I could misunderstand it. It seems like the generic utilities should be in the base class. Specific utilities would go in subclasses...

Reply:
We actually started this file in the machine learning module. At the moment, modify_for_projected_storage and these other functions are only used by persistence. We plan to use modify_for_projected_storage and these same functions when we add the module for integrating RFC forecasts and whenever we add ML. Nels suggested having utilities that would be generic to multiple types of reservoirs in a separate utilities file to keep the base class from getting too bloated.

Reply:
OK. That's fine.